src/tif_model_iterator/tif_kernel_iterator.py

```diff
@@ -187,16 +187,46 @@ def _create_pixel_coordinate_dataframe(
             [y for y in range(0, data.shape[2])] for x in range(0, data.shape[1])
         ]
 
-        rd_x, rd_y = rasterio.transform.xy(
-            self.dataset.transform, x_coordinates, y_coordinates
+        rd_x_ul, rd_y_ul = rasterio.transform.xy(
+            self.dataset.transform, x_coordinates, y_coordinates, offset="ul"
         )
-        data = np.append(data, rd_x).reshape([z_shape + 1, x_shape, y_shape])
-        data = np.append(data, rd_y).reshape([z_shape + 2, x_shape, y_shape])
+
+        rd_x_ur, rd_y_ur = rasterio.transform.xy(
+            self.dataset.transform, x_coordinates, y_coordinates, offset="ur"
+        )
+
+        rd_x_ll, rd_y_ll = rasterio.transform.xy(
+            self.dataset.transform, x_coordinates, y_coordinates, offset="ll"
+        )
+
+        rd_x_lr, rd_y_lr = rasterio.transform.xy(
+            self.dataset.transform, x_coordinates, y_coordinates, offset="lr"
+        )
+
+        data = np.append(data, rd_x_ul).reshape([z_shape + 1, x_shape, y_shape])
+        data = np.append(data, rd_y_ul).reshape([z_shape + 2, x_shape, y_shape])
+        data = np.append(data, rd_x_ur).reshape([z_shape + 3, x_shape, y_shape])
+        data = np.append(data, rd_y_ur).reshape([z_shape + 4, x_shape, y_shape])
+        data = np.append(data, rd_x_ll).reshape([z_shape + 5, x_shape, y_shape])
+        data = np.append(data, rd_y_ll).reshape([z_shape + 6, x_shape, y_shape])
+        data = np.append(data, rd_x_lr).reshape([z_shape + 7, x_shape, y_shape])
+        data = np.append(data, rd_y_lr).reshape([z_shape + 8, x_shape, y_shape])
+
         data = data.reshape(-1, x_shape * y_shape).transpose()
 
         df = pd.DataFrame(
             data,
-            columns=self.column_names + ["rd_x", "rd_y"],
+            columns=self.column_names
```

Comment:
Suggestion: I would update the actual self.column_names value, instead of adding extra names here. Otherwise it might be quite confusing.